core/domain/entities/editor/blocks/widgets/EventAttendees.php

```diff
@@ -0,0 +1,163 @@
+<?php
+
+namespace EventEspresso\core\domain\entities\editor\blocks\widgets;
+
+use DomainException;
+use EE_Error;
+use EEM_Registration;
+use EventEspresso\core\domain\entities\editor\Block;
+use EventEspresso\core\domain\entities\editor\blocks\CoreBlocksAssetManager;
+use EventEspresso\core\domain\entities\shortcodes\EspressoEventAttendees;
+use EventEspresso\core\exceptions\InvalidDataTypeException;
+use EventEspresso\core\exceptions\InvalidInterfaceException;
+use InvalidArgumentException;
+
+/**
+ * Class EventAttendees
+ * Returns a list of people that have registered for the specified event
+ *
+ * @package EventEspresso\core\domain\entities\editor\blocks\common
+ * @author  Brent Christensen
+ * @since   $VID:$
+ */
+class EventAttendees extends Block
+{
+
+    const BLOCK_TYPE = 'widgets-event-attendees';
+
+    /**
+     * @var EspressoEventAttendees $shortcode
+     */
+    protected $shortcode;
+
+
+    /**
+     * EventAttendees constructor.
+     *
+     * @param CoreBlocksAssetManager $block_asset_manager
+     * @param EspressoEventAttendees $shortcode
+     */
+    public function __construct(CoreBlocksAssetManager $block_asset_manager, EspressoEventAttendees $shortcode)
+    {
+        parent::__construct($block_asset_manager);
+        $this->shortcode = $shortcode;
+    }
+
+
+    /**
+     * Perform any early setup required by the block
+     * including setting the block type and supported post types
+     *
+     * @return void
+     */
+    public function initialize()
+    {
+        $this->setBlockType(self::BLOCK_TYPE);
+        $this->setSupportedRoutes(
+            array(
+                'EventEspresso\core\domain\entities\route_match\specifications\admin\EspressoEventEditor',
+                'EventEspresso\core\domain\entities\route_match\specifications\admin\WordPressPostsEditor',
+                'EventEspresso\core\domain\entities\route_match\specifications\frontend\EspressoBlockRenderer',
+            )
+        );
+        $this->setAttributes(
+            array(
+                'eventId'            => array(
+                    'type'    => 'number',
+                    'default' => 0,
```

Comment:
Related to my previous comment about the block being used in the Event Editor, I wonder if this is where you can set the default conditionally?

Reply:
yes I believe that is correct, but I'll also investigate some of the WP core blocks to see what they are doing